solutions/src/main/java/ru/sberbank/school/task10/ScalableThreadPool.java

```diff
@@ -0,0 +1,139 @@
+package ru.sberbank.school.task10;
+
+import lombok.NonNull;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+
+public class ScalableThreadPool implements ThreadPool {
+    private final Queue<Runnable> taskQueue;
+
+    private final int min;
+    private final int max;
+    private volatile int workingThreads = 0;
+    private final List<ThreadWorker> threads;
+    private final List<ThreadWorker> excessiveThreads;
+    private volatile int freeId;
+
+    public ScalableThreadPool(int min, int max) {
+        if (min <= 0 || max <= 0 || min > max) {
+            throw new IllegalArgumentException();
+        }
+        this.taskQueue = new LinkedList<>();
+        this.min = min;
+        this.max = max;
+        this.freeId = min + 1;
+        this.threads = new ArrayList<>(min);
+        this.excessiveThreads = new ArrayList<>(max-min);
```

Comment:
WhitespaceAround: &#39;-&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-375315140>*